### PR TITLE
Additional janno columns

### PR DESCRIPTION
**Pull request by nevrome**
This PR implements [a mechanism](https://stackoverflow.com/questions/74598030/haskell-cassava-data-csv-carry-along-additional-columns) to carry along additional, unspecified variables in .janno files. This is a long-desired (see #109) key feature to improve the flexibility of .janno files even more.

For now this is an early draft, which needs thorough testing to make sure this functionality does not break any workflow and works exactly as expected.

### CODE REVIEW

**Comment by nevrome** [comment]
I think this works now. But it's a pretty significant change, so I'm a bit insecure if I might have forgotten about something in the bigger scheme of things.  For example the way I ignored this new functionality for JSON conversion.

Could you please review, @stschiff?

**Comment by stschiff** [comment]
OK, sorry for being late on this, but not forgotten.

File: src/Poseidon/Janno.hs [hunk 1]
```diff
@@ -663,64 +739,37 @@ instance Csv.ToNamedRecord JannoRow where
...
-
 jannoHeaderString :: [String]
 jannoHeaderString = map Bchs.unpack jannoHeader
 
+makeHeaderWithAdditionalColumns :: [JannoRow] -> Csv.Header
+makeHeaderWithAdditionalColumns ms =
+    V.fromList $ jannoHeader ++ sort (HM.keys (HM.unions (map (getCsvNR . jAdditionalColumns) ms)))
+
+concatJannos :: [[JannoRow]] -> [JannoRow]
+concatJannos = foldl' combineTwoJannos []
```
Comment (stschiff) [inline src/Poseidon/Janno.hs:756]: Nice!

**Review by stschiff — APPROVED** [review]
Amazing job. Thanks! I've gone through the code changes and looks all good to me. I have so little capacity right now that I'm afraid I can't test it, but I think we should just go for it. You've added the tests and if there is a problem we'll fix it in a patch. So good to merge!

**Comment by nevrome** [comment]
Alright - thanks! I'll merge but do some more testing and additional, minor changes in #209 before a new release.

**Pull request merged**